### PR TITLE
Limit payment method list builder to user in route

### PR DESCRIPTION
**Pull request by bradjones1**
Unlike most other list builders, which are administrative, this one is context-specific to the route.

### CODE REVIEW

**Comment by bradjones1** [comment]
https://www.drupal.org/node/2776417

File: modules/payment/src/PaymentMethodListBuilder.php [hunk 1]
```diff
@@ -22,6 +23,16 @@ public function getFormId() {
     return 'commerce_payment_methods';
   }
 
+  protected function getEntityIds() {
```
Comment (bojanz) [inline modules/payment/src/PaymentMethodListBuilder.php:28]: The parent method also has support for paging, we should recopy that code:
    // Only add the pager if a limit is specified.
    if ($this->limit) {
      $query->pager($this->limit);
    }

We also need a docblock with an inheritdoc inside.

File: modules/payment/src/PaymentMethodListBuilder.php [hunk 2]
```diff
@@ -22,6 +23,16 @@ public function getFormId() {
...
+    /** @var RouteMatchInterface $route */
```
Comment (borisson) [inline modules/payment/src/PaymentMethodListBuilder.php]: Not sure how strict @bojanz is with coding standards, but this should be the FQN: 
`/** @var \Drupal\Core\Routing\RouteMatchInterface $route */`

File: modules/payment/src/PaymentMethodListBuilder.php [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Entity\EntityListBuilder;
+use Drupal\Core\Routing\RouteMatchInterface;
```
Comment (borisson) [inline modules/payment/src/PaymentMethodListBuilder.php]: Is this use still needed now that the `@var` no longer uses the imported interface?

**Comment by borisson** [comment]
This looks great now.

**Comment by bojanz** [comment]
Squashed this into payment in preparation for merge. Gave you commit credit on a different commit in return: fdc891093fe4006698b4ab8c6e467a7f85f183aa

**Pull request closed without merging**